datajunction-server/datajunction_server/api/helpers.py

```diff
@@ -871,6 +856,49 @@ async def query_event_stream(  # pylint: disable=too-many-arguments
         await asyncio.sleep(stream_delay)  # pragma: no cover
 
 
+def build_sql_for_dj_query(  # pylint: disable=too-many-arguments,too-many-locals
+    session: Session,
+    query: str,
+    engine_name: Optional[str] = None,
+    engine_version: Optional[str] = None,
+) -> Tuple[TranslatedSQL, Engine, Catalog]:
+    """
+    Build SQL for multiple metrics. Used by both /sql and /data endpoints
+    """
+
+    query_ast, metrics = build_dj_query(session, query)
+
+    leading_metric_node = metrics[0]
+    available_engines = leading_metric_node.current.catalog.engines
+
+    # Check if selected engine is available, or if none is provided, select the fastest
```

Comment:
I guess technically this comment isn't true, right? We don't actually rank the engines by speed, we just select the first one available. At some point we might want to include a ranking of engines that users can provide manually.

Reply:
I think this is from copy pasta so we need to change it elsewhere as well